### PR TITLE
feat: check links via linkinator

### PR DESCRIPTION
**Pull request by mshanemc**
check markdown files for broken links
1. top-level of the project that are not CHANGELOG.md
1. and everything in `/messages`

[@W-14543815@](https://gus.lightning.force.com/a07EE00001f4nnFYAQ)

### CODE REVIEW

**Comment by iowillhoit** [comment]
I wonder if we should come up with a way to skip this if needed. Maybe with an environment variable? I am imagining this exiting on a false positives (website down) and preventing a release. You could force merge a PR but you would not have any NUTs run if the unit tests failed because of linkinator. [Example GHA](https://github.com/salesforcecli/plugin-info/blob/49104b70edd5ad14998e43629654f2686a28b08f/.github/workflows/test.yml#L18-L19)

File: utils/sf-config.js [hunk 1]
```diff
@@ -47,14 +47,20 @@ const PACKAGE_DEFAULTS = {
       files: ['src/**/*.ts', 'test/**/*.ts', 'messages/**', '**/.eslint*', '**/tsconfig.json'],
       output: [],
     },
+    'link-check': {
+      command:
+        'linkinator "./!(CHANGELOG).md" "messages/**/*.md" --markdown --retry --directory-listing --verbosity error',
```
[the remark this answers is not in the document]
Comment (mshanemc) [reply utils/sf-config.js:52]: commit that!

**Comment by mshanemc** [comment]
> I wonder if we should come up with a way to skip this if needed. Maybe with an environment variable? I am imagining this exiting on a false positives (website down) and preventing a release. You could force merge a PR but you would not have any NUTs run if the unit tests failed because of linkinator. [Example GHA](https://github.com/salesforcecli/plugin-info/blob/49104b70edd5ad14998e43629654f2686a28b08f/.github/workflows/test.yml#L18-L19)

Yeah, so the GHA could read from GHA variables to skip itself?  Good idea, commit that, too.

**Comment by iowillhoit** [comment]
Ok @mshanemc this is ready for you to take a look at. I made a couple more changes to this. 

- Changed `^test` to `test/`. Despite what their docs say, I could not get this regex to work.
  - The downside of this is that it will also skip nested `test` directories. For example `messages/test/foo.md`
- Added `%s` like we use in `plugin-info`
- Added `link-check` to the `scripts` so you can run it on its own
- Added a cross-os way of skipping the link-check. See screenshot
Screenshot 2023-12-05 at 3 20 28 PM

**Comment by iowillhoit** [comment]
Hold up a bit on this, the env var is not working in CI. Trying a couple things.

**Pull request merged**